compiler/src/dotty/tools/dotc/plugins/Plugins.scala

```diff
@@ -119,8 +119,12 @@ trait Plugins {
     // plugin-specific options.
     // The user writes `-P:plugname:opt1,opt2`, but the plugin sees `List(opt1, opt2)`.
     def options(plugin: Plugin): List[String] = {
-      def namec = plugin.name + ":"
-      ctx.settings.pluginOptions.value filter (_ startsWith namec) map (_ stripPrefix namec)
+      val namec = plugin.name + ":"
+      ctx.settings.pluginOptions.value
+        .filter: option =>
+          option.startsWith(namec)
+        .flatMap: option => 
+          option.stripPrefix(namec).split(",")
```

Comment:
If there is a comma in the value in the opt, we will get an error value here.

Reply:
ah yea, good call. I think before changing anything I'd like to get some input from maintainers if it's actually desired to have this ability to pass in multiple separated values at one, or they want one value per flag passed in.

Reply:
ah yea, good call. I think before changing anything I'd like to get some input from maintainers if it's actually desired to have this ability to pass in multiple separated values at one, or they want one value per flag passed in.

Reply:
this is scala 2 behavior, as I confirmed earlier. The protocol for `init` is slightly different in scala 3, in scala 2 it lets the plugin opt-out of the run, in scala 3 it returns phases, maybe it can opt-out by return no phases? My other digression is that I opened a scala 2 ticket suggesting to change plugin options to "first class citizens" but I think I was dissuaded. The prevailing convention is ugly but legacy.

Reply:
huh, ok, so @som-snytt do you have any idea if we're meant to keep this behavior here?

Reply:
In considering the tradeoff between usability vs. flexibility:

1. There is the possibility of using `,` as a valid part of plugin arguments, but it seems to be a rare case.
2. Meanwhile, it seems to be trivial for plugin authors to handle `,` in the arguments and it's more flexible.

It seems either choice is reasonable. 

For compatibility, changing the behavior might potentially break some existing plugins. Maybe we can keep the existing behavior for the sake of compatibility (update the misleading comment though)? /cc: @som-snytt

Reply:
Yes, I think it's better not to poke the bear.

Reply:
Makes sense. I've reverted the change and then just updated the scaladocs to reflect this.